### PR TITLE
[Example] Multi-GPU training by graph sampling

### PR DESCRIPTION
**Pull request by jermainewang**
## Description

Example codes for training GNNs on a single machine but multiple GPUs by sampling from a single graph data.

The example includes three models: GCN, GraphSAGE and GAT.

Results
---

Testbed: one AWS EC2 p3.16xlarge instance, 8 V100 GPUs, 64 VCPUs

**GCN:** `python main.py --gpu=0 --model=gcn --num-workers=4`

Accuracy: 93.87%

| | Speed (K samples/sec) |
| --- | --- |
| 1 GPU | 10.8 |
| 2 GPU | 24.1 |
| 4 GPU | 41.5 |
| 8 GPU | 69.1 |

**GraphSAGE:** `python main.py --gpu=0 --model=sage --num-workers=4`

Accuracy: 95.04%

| | Speed (K samples/sec) |
| --- | --- |
| 1 GPU | 10.8 |
| 2 GPU | 23.3 |
| 4 GPU | 40.5 |
| 8 GPU | 59.5 |

**GAT:** `python main.py --gpu=0 --model=gat --num-workers=4 --lr=0.01`

Accuracy: 93.7%

| | Speed (K samples/sec) |
| --- | --- |
| 1 GPU | 5.11 |
| 2 GPU | 10.4 |
| 4 GPU | 19.9 |
| 8 GPU | 33.1 |

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

File: examples/pytorch/sampling/multi-gpu/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+Multi-GPU training by graph sampling
+===
+
+The example trains GNN models on a single machine but multiple GPUs by sampling from a single graph data. The setting is suitable for the case where the graph is too large to fit in one GPU memory.
```
Comment (futurely) [inline examples/pytorch/sampling/multi-gpu/README.md]: The graph is in the CPU memory. The major benefit of using multiple gpus is to speed up running.

Comment (yzh119) [reply examples/pytorch/sampling/multi-gpu/README.md]: OK, to be more specific:
1. sampling is designed for the case where graph is too large to fit in one GPU memory.
2. multi-gpu setting is for speedup training.

**Comment by 510443664** [comment]
Hi,
It really helps a lot.
Would you help to modify the examples for R-GCN?
Thanks.

**Comment by jermainewang** [comment]
Most codes have been merged in PR #1333

**Pull request closed without merging**